pkg/providers/cloudstack/cloudstack.go

```diff
@@ -166,10 +166,11 @@ func (p *cloudstackProvider) ValidateNewSpec(ctx context.Context, cluster *types
 		return err
 	}
 
-	datacenter := p.datacenterConfig
+	prevDatacenter.SetDefaults()
 
-	oSpec := prevDatacenter.Spec
-	nSpec := datacenter.Spec
+	if err = clusterSpec.CloudStackDatacenter.ValidateUpdate(prevDatacenter); err != nil {
```

Comment:
currently `ValidateUpdate` is only used for webhook. It makes sense to call it in CLI here since all the validations are same (as for now). We should be careful about adding new validation logic in the webhook since it also affects CLI.